lib/trino-collect/src/test/java/io/trino/collect/cache/TestEvictableCache.java

```diff
@@ -275,6 +279,60 @@ private static Integer newInteger(int value)
         return newInteger;
     }
 
+    /**
+     * Test that the loader is invoked only once for concurrent invocations of {{@link LoadingCache#get(Object, Callable)} with equal keys.
+     * This is a behavior of Guava Cache as well. While this is necessarily desirable behavior (see
+     * <a href="https://github.com/trinodb/trino/issues/11067">https://github.com/trinodb/trino/issues/11067</a>),
+     * the test exists primarily to document current state and support discussion, should the current state change.
+     */
+    @Test(timeOut = TEST_TIMEOUT_MILLIS)
+    public void testConcurrentGetWithCallableShareLoad()
```

Comment:
We should look into jcstress (https://github.com/openjdk/jcstress) for testing concurrency (for later, not as part of this PR).